presentation-api/controlling-ua/PresentationConnection_onclosed-manual.html

```diff
@@ -21,28 +21,36 @@
 <script>
   setup({explicit_timeout: true});
   var startPresentation = function () {
+
     document.getElementById('presentBtn').disabled = true;
-    async_test(function(t) {
+
+    promise_test(function (t) {
       var request = new PresentationRequest(presentationUrls);
-      request.start()
-        .then(function(connection) {
+
+      return request.start()
+        .then(function (connection) {
 
           // Enable timeout again, cause no user action is needed from here.
-          t.step_timeout(function() {
+          t.step_timeout(function () {
               t.force_timeout();
               t.done();
           }, 5000);
 
           assert_true(connection instanceof PresentationConnection, 'the connection is setup');
-          connection.onclose = t.step_func_done(function(evt) {
+
+          connection.onclose = function (evt) {
             assert_equals(evt.type, "close");
             assert_equals(connection.state, "closed");
-          });
+          };
+
           connection.close();
+
       })
-      .catch(function(ex) {
+      .catch(function (ex) {
```

Comment:
You don't need this catch in a promise_test. The promise_test will fail if the promise passed to it rejects.